front/src/applications/operationalStudies/components/Scenario/AddOrEditScenarioModal.tsx

```diff
@@ -9,86 +9,86 @@ import { ModalContext } from 'common/BootstrapSNCF/ModalSNCF/ModalProvider';
 import SelectImprovedSNCF from 'common/BootstrapSNCF/SelectImprovedSNCF';
 import TextareaSNCF from 'common/BootstrapSNCF/TextareaSNCF';
 import InfraSelectorModal from 'common/InfraSelector/InfraSelectorModal';
-import { deleteRequest, get, patch, post } from 'common/requests';
 import { useTranslation } from 'react-i18next';
 import { FaPencilAlt, FaPlus, FaTrash } from 'react-icons/fa';
 import { GiElectric } from 'react-icons/gi';
 import { MdDescription, MdTitle } from 'react-icons/md';
 import { useDispatch, useSelector } from 'react-redux';
 import { useNavigate } from 'react-router-dom';
-import { setSuccess } from 'reducers/main';
+import { setFailure, setSuccess } from 'reducers/main';
 import { updateScenarioID } from 'reducers/osrdconf';
 import { getInfraID, getProjectID, getStudyID } from 'reducers/osrdconf/selectors';
-import { ScenarioListResult } from 'common/api/osrdEditoastApi';
-import {
-  ELECTRICAL_PROFILE_SET_URI,
-  PROJECTS_URI,
-  SCENARIOS_URI,
-  STUDIES_URI,
-} from '../operationalStudiesConsts';
+import { ScenarioListResult, osrdEditoastApi, ScenarioRequest } from 'common/api/osrdEditoastApi';
+import { sortBy } from 'lodash';
 
 const scenarioTypesDefaults = {
   name: '',
   description: '',
   infra: undefined,
 };
 
-type Props = {
+type AddOrEditScenarioModalProps = {
   editionMode?: boolean;
   scenario?: ScenarioListResult;
   getScenarioTimetable?: (v: boolean) => void;
 };
 
+type SelectOptionsType = { key: number | undefined; value: string };
+
 export default function AddOrEditScenarioModal({
   editionMode = false,
   scenario,
   getScenarioTimetable,
-}: Props) {
+}: AddOrEditScenarioModalProps) {
   const { t } = useTranslation('operationalStudies/scenario');
   const { closeModal } = useContext(ModalContext);
+
+  const [deleteScenarioRTK] =
+    osrdEditoastApi.useDeleteProjectsByProjectIdStudiesAndStudyIdScenariosScenarioIdMutation({});
+  const [patchScenario] =
+    osrdEditoastApi.usePatchProjectsByProjectIdStudiesAndStudyIdScenariosScenarioIdMutation({});
+  const [postScenario] =
+    osrdEditoastApi.usePostProjectsByProjectIdStudiesAndStudyIdScenariosMutation({});
+
   const noElectricalProfileSetOption = {
     key: undefined,
     value: t('noElectricalProfileSet'),
   };
   const [currentScenario, setCurrentScenario] = useState<ScenarioListResult>(
     scenario || scenarioTypesDefaults
   );
+  const { data: electricalProfils, isError } = osrdEditoastApi.useGetElectricalProfileSetQuery();
```

Comment:
Set electricalProfils to [] by default to avoid bugs if there is an error. Also, you can use selectFromResult to remove the variable `electricalProfileSetOptions`
```suggestion
  const { data: electricalProfilOptions = [], isError } = osrdEditoastApi.useGetElectricalProfileSetQuery({
    selectFromResult: (response) => {
      #(code of electricalProfileSetOptions)
    }
  });
```

Reply:
Done !!!!